### PR TITLE
Fix a crash issue on Android "Fatal Exception: java.lang.IllegalStateException"

### PR DESCRIPTION
**Pull request by thanakij**
Hi,

Our team have experience a crash issue reported by Crashlytics saying "_Fatal Exception. This YouTubePlayer has been released (**YouTubePlayerController.java:315**)_"

We are using latest commit [#1c2d9dc](https://github.com/inProgress-team/react-native-youtube.git#1c2d9dc) (at the time of writing) for our package.json.

I have tried to strengthen the code with the "try ... catch" block and
would like to submit a pull request for review.

Thanks!

### CODE REVIEW

**Comment by chrismpalmer** [comment]
Did this work for you? I was reviewing your code. I was heading down the same lines. Just wondering if it's working?

**Comment by thanakij** [comment]
> Did this work for you? I was reviewing your code. I was heading down the same lines. Just wondering if it's working? @thanakij

There are no Crashlytics report of the same errors so far.

However, I have found one new issue, "_No view found for id 0x36e7 (unknown) for fragment VideoFragment{664a5dc #0 id=0x36e7}. (**OS: FragmentManager.java:1004**)_".

This StackOverflow [discussion](https://stackoverflow.com/questions/7508044/android-fragment-no-view-found-for-id) reveals that it may relate to my changes. The error is likely from _fragmentManager.beginTransaction().add(getId(), mVideoFragment)_ and for some reason such view cannot be found.

**Comment by DanielLuu** [comment]
@thanakij does your fix prevent the new issue?

**Comment by thanakij** [comment]
No, like discussed earlier, the fix contain new issue. I believe this is because the root cause is not resolved. Eventually, we simply get a different error messages when crashing (but it is sure less often). Let's look at #336. It also tried to tackle the same original issue.

**Comment by thanakij** [comment]
For anybody who visits this, I have decided to drop this react-native-youtube dependency from my project as this lib is pretty much dead and unmaintained.

Next, I have replaced it with WebView for now.

```
<WebView
    style={{ flex: 1, height: VIDEO_HEIGHT }}
    javaScriptEnabled={true}
    source={{ uri: uri }}
/>
```

It works well so far. In addition, you may have to use [react-native-android-fullscreen-webview-video](https://github.com/teamairship/react-native-android-fullscreen-webview-video) to enable "fullscreen" mode for Android.

**Comment by davidohayon669** [comment]
@thanakij maybe your issue has something to do with #357? Please offer a conclusive PR so I can understand the issue and we can fix this

**Comment by thanakij** [comment]
@davidohayon669 #357 is also using _commitAllowingStateLoss()_, but like what I experienced with my own pull request, it probably would result in another issue, "No view found".

For _commit()_, if there is a state error, it will throw an exception. However, by using _commitAllowingStateLoss()_, if there is a state error, the error is still there but Android will not throw an exception and we wouldn't know.

In short, it was meant for a quick fix and [many](https://www.androiddesignpatterns.com/2013/08/fragment-transaction-commit-state-loss.html) suggested to use this method only for a last resort.

I don't know what is the real root cause for this IllegalStateException but something like #336 looks interesting to me because it added _mPlayerAvailable_ for safety..
